lib/rust/json-to-struct/src/lib.rs

```diff
@@ -0,0 +1,366 @@
+const BOILERPLATE: &str = "
+/// Error containing the path to the field and the reason of the error.
+#[derive(Debug, Clone)]
+pub struct Error {
+    pub path: String,
+    pub reason: String,
+}
+
+impl Error {
+    /// Constructor.
+    pub fn new(path: impl Into<String>, reason: impl Into<String>) -> Self {
+        let path = path.into();
+        let reason = reason.into();
+        Self { path, reason }
+    }
+
+    /// Change the path of this error.
+    pub fn with_path(self, path: impl Into<String>) -> Self {
+        let path = path.into();
+        Self { path, ..self }
+    }
+
+    /// Convert the error to a human-readable form.
+    pub fn display(&self) -> String {
+        format!(\"{}: {}\", self.path, self.reason)
+    }
+}
+
+/// Try setting a boolean value. If the value is not a boolean, returns an error.
+pub fn set_bool(target: &mut bool, value: &str, name: &str) -> Option<Error> {
+    let parsed_value = match value {
+        \"false\" => Some(false),
+        \"true\" => Some(true),
+        \"0\" => Some(false),
+        \"1\" => Some(true),
+        \"enabled\" => Some(true),
+        \"disabled\" => Some(false),
+        \"on\" => Some(true),
+        \"off\" => Some(false),
+        _ => None
+    };
+    match parsed_value {
+        Some(value) => {
+            *target = value;
+            None
+        },
+        None => Some(Error::new(name, format!(\"Expected a boolean, got '{}'\", value)))
+    }
+}
+
+/// Try setting a number value. If the value is not a number, returns an error.
+pub fn set_number(target: &mut f64, value: &str, name: &str) -> Option<Error> {
+    if let Ok(number) = value.parse::<f64>() {
+        *target = number;
+        None
+    } else {
+        Some(Error::new(name, format!(\"Expected a number, got '{}'\", value)))
+    }
+}
+
+/// Try setting a string value. If the value is not a string, returns an error.
+pub fn set_string(target: &mut String, value: &str, name: &str) -> Option<Error> {
+    *target = value.to_owned();
+    None
+}
+
+/// Allows setting values by names of the original JSON fields. The name can contain dots to refer
+/// to nested fields.
+#[allow(missing_docs)]
+trait Setter {
+    fn set(&mut self, name: &str, value: String) -> Option<Error>;
+}
+";
+
+
+
+// ======================
+// === JSON To Struct ===
+// ======================
+
+/// Converts the provided JSON files into a Rust struct. If more than one file is provided, the
+/// files will be merged. In case any keys will overlap, it will panic. Supports all JSON types but
+/// arrays.
+///
+/// For example, for the following JSON file (`test.json`):
+/// ```text
+/// {
+///   "test": {
+///     "nested": {
+///       "field1": 1
+///     },
+///     "field2": "value"
+///   }
+/// }
+/// ```
+/// The macro call `json_to_struct("test.json")` will generate the following Rust code:
+///
+/// ```text
+/// <BOILERPLATE> // See the `BOILERPLATE` constant to learn more.
```

Comment:
You can make somewhere a link [`BOILERPLATE`] and make the constant public: it will look nice in generated docs.